app/routes/individual_response.py

```diff
@@ -248,7 +250,12 @@ def individual_response_text_message_confirmation(schema, questionnaire_store):
     if request.method == "POST":
         return redirect(url_for("questionnaire.get_questionnaire"))
 
-    mobile_number = url_safe_serializer().loads(request.args.get("mobile_number"))
+    if not (request_mobile_number := request.args.get("mobile_number")):
+        raise BadRequest
+    try:
+        mobile_number = url_safe_serializer().loads(request_mobile_number)
+    except BadSignature:
```

Comment:
This could be BadData if we wanted to cover the base exception for itsdangerous, although I am not convinced that we need any of the others, and if we did they should be 500

Reply:
I agree, I think if it's `BadData` it's more likely ti be something wrong on the server side than the client.